### DISCUSSION TITLE
Add "(un)select all" option for dropdown lists

### DISCUSSION THREAD

**Discussion by XiangRongLin** [RedashCommunity/redash#34] [Ideas]
For (query based) dropdown lists, if there are a lot of options, (un)selecting each one manually is very slow. Thus a "select all" and "unselect all" option would be helpfull.

I took a look in the code. the `Select` Component is based on antd, which itself does not support this feature. It was requested [there](https://github.com/ant-design/ant-design/issues/7155) too, but ultimately not implemented.
Community members have provided a custom solution, which I implemented [here](https://github.com/XiangRongLin/redash/commit/9c7bde91237191efd2c06ae73ce4b607af6e1c7e). 
It works, but still has an error that I do not quite understand, since I have little knowledge about react (line 38 and 44).

**Comment by gaecoli** [comment]
> For (query based) dropdown lists, if there are a lot of options, (un)selecting each one manually is very slow. Thus a "select all" and "unselect all" option would be helpfull.
> 
> I took a look in the code. the `Select` Component is based on antd, which itself does not support this feature. It was requested [there](https://github.com/ant-design/ant-design/issues/7155) too, but ultimately not implemented. Community members have provided a custom solution, which I implemented [here](https://github.com/XiangRongLin/redash/commit/9c7bde91237191efd2c06ae73ce4b607af6e1c7e). It works, but still has an error that I do not quite understand, since I have little knowledge about react (line 38 and 44).

This is a good feature.

**Comment by gaecoli** [comment]
> For (query based) dropdown lists, if there are a lot of options, (un)selecting each one manually is very slow. Thus a "select all" and "unselect all" option would be helpfull.
> 
> I took a look in the code. the `Select` Component is based on antd, which itself does not support this feature. It was requested [there](https://github.com/ant-design/ant-design/issues/7155) too, but ultimately not implemented. Community members have provided a custom solution, which I implemented [here](https://github.com/XiangRongLin/redash/commit/9c7bde91237191efd2c06ae73ce4b607af6e1c7e). It works, but still has an error that I do not quite understand, since I have little knowledge about react (line 38 and 44).

I merged it. Thank you!

**Reply by justinclift** [reply]
This bit worries me:

> It works, but still has an error that I do not quite understand ...

Does that mean it works in the version that was merged, or it still has an error? :smile:

**Reply by XiangRongLin** [reply]
It still contains the error:
`TS2345: Argument of type 'ReactText[]' is not assignable to parameter of type 'string'.`

**Reply by XiangRongLin** [reply]
> I merged it. Thank you!

Waiting for the docker image to use it 🍿

**Reply by gaecoli** [reply]
> It still contains the error: `TS2345: Argument of type 'ReactText[]' is not assignable to parameter of type 'string'.`

I will fix it.

**Reply by XiangRongLin** [reply]
@gaecoli Let me know when you merge it. I'm interested in how to do it, since I couldn't figure it out

**Reply by gaecoli** [reply]
> @gaecoli Let me know when you merge it. I'm interested in how to do it, since I couldn't figure it out

Ok! This week, i will fix it.
